llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1028,79 +1055,113 @@ bool RISCVVLOptimizer::isCandidate(const MachineInstr &MI) const {
   return true;
 }
 
-bool RISCVVLOptimizer::checkUsers(const MachineOperand *&CommonVL,
-                                  MachineInstr &MI) {
+std::optional<MachineOperand>
+RISCVVLOptimizer::getVLForUser(MachineOperand &UserOp) {
```

Comment:
Nit, I think a better name for this might be something like "getMinimumVLForUser". 

Since for regular vector operands the minimum VL we can reduce to is the VL operand but for scalar vector operands it's one.

Reply:
updated